mmv1/api/resource.rb

```diff
@@ -435,6 +439,34 @@ def decoder?
       !@transport&.decoder.nil?
     end
 
+    def add_labels_related_fields(props)
+      props.each do |p|
+        if p.is_a? Api::Type::KeyValueLabels
+          props << build_effective_labels_field('labels', p.field_min_version)
+        elsif p.is_a? Api::Type::KeyValueAnnotations
+          props << build_effective_labels_field('annotations', p.field_min_version)
+        elsif (p.is_a? Api::Type::NestedObject) && !p.all_properties.nil?
+          p.properties = add_labels_related_fields(p.all_properties)
+        end
+      end
+      props
+    end
+
+    def build_effective_labels_field(name, min_version)
+      description = "All of #{name} (key/value pairs)\
```

Comment:
Should this be consistent w/ handwritten fields?

Reply:
Yes, it should. Will use this description for all of the effective_labels/effective_annotations field.